mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ResourceClustersHostManagerActor.java

```diff
@@ -417,4 +417,29 @@ private static Optional<String> validateClusterSpec(ProvisionResourceClusterRequ
 
         return Optional.empty();
     }
+
+    /**
+     * Validates whether the SkuTypeSpec object has valid size specifications.
+     * Checks whether each resource component (CPU cores, disk storage, memory, and network speed) is greater than or equal to 1;
+     * If NULL size is detected, it validates each component individually.
+     *
+     * @param sku instance of SkuTypeSpec to be validated.
+     * @return Boolean value indicating whether the `sku` parameter's size specs are valid.
+     */
+    private static boolean isSkuSizeValid(SkuTypeSpec sku) {
+        // If size is null, check each individual attribute
+        if (sku.getSize() == null) {
+            return sku.getCpuCoreCount() >= 1
+                && sku.getDiskSizeInMB() >= 1
+                && sku.getMemorySizeInMB() >= 1
+                && sku.getNetworkMbps() >= 1;
+        }
+
+        // If size has been defined, validate its components
+        MachineDefinition machineDefinition = sku.getSize().getMachineDefinition();
+        return machineDefinition.getCpuCores() >= 1
+            && machineDefinition.getDiskMB() >= 1
+            && machineDefinition.getMemoryMB() >= 1
+            && machineDefinition.getNetworkMbps() >= 1;
+    }
```

Comment:
Can we do this validation when the SkuTypeSpec object is being created? It feels like we can avoid doing this check everywhere.